example_unrolling_service/example_without_bazel.py

```diff
@@ -0,0 +1,190 @@
+It is equivalent in behavior to the example.py script in this directory.
+"""
+import os
+import subprocess
+from pathlib import Path
+from typing import Iterable
+
+import compiler_gym
+from compiler_gym.datasets import Benchmark, Dataset
+from compiler_gym.envs.llvm.llvm_benchmark import get_system_includes
+from compiler_gym.spaces import Reward
+from compiler_gym.third_party import llvm
+from compiler_gym.util.registration import register
+from compiler_gym.util.commands import run_command
+
+UNROLLING_PY_SERVICE_BINARY: Path = Path(
+    "example_unrolling_service/service_py/example_service.py"
+)
+assert UNROLLING_PY_SERVICE_BINARY.is_file(), "Service script not found"
+
+BENCHMARKS_PATH: Path = Path("benchmarks")
+
+NEURO_VECTORIZER_HEADER: Path = Path(
+    "third_party/neuro-vectorizer/header.h"
+)
+
+
+class RuntimeReward(Reward):
+    """An example reward that uses changes in the "runtime" observation value
+    to compute incremental reward.
+    """
+
+    def __init__(self):
+        super().__init__(
+            id="runtime",  # name="runtime", #FIXME
+            observation_spaces=["runtime"],
+            default_value=0,
+            default_negates_returns=True,
+            deterministic=False,
+            platform_dependent=True,
+        )
+        self.baseline_runtime = 0
+
+    def reset(self, benchmark: str, observation_view):
+        del benchmark  # unused
+        self.baseline_runtime = observation_view["runtime"]
+
+    def update(self, action, observations, observation_view):
+        del action  # unused
+        del observation_view  # unused
+        return float(self.baseline_runtime - observations[0]) / self.baseline_runtime
+
+
+class SizeReward(Reward):
+    """An example reward that uses changes in the "size" observation value
+    to compute incremental reward.
+    """
+
+    def __init__(self):
+        super().__init__(
+            id="size",  # name="size", #FIXME
+            observation_spaces=["size"],
+            default_value=0,
+            default_negates_returns=True,
+            deterministic=False,
+            platform_dependent=True,
+        )
+        self.baseline_size = 0
+
+    def reset(self, benchmark: str, observation_view):
+        del benchmark  # unused
+        self.baseline_runtime = observation_view["size"]
+
+    def update(self, action, observations, observation_view):
+        del action  # unused
+        del observation_view  # unused
+        return float(self.baseline_size - observations[0]) / self.baseline_size
+
+
+class UnrollingDataset(Dataset):
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            name="benchmark://unrolling-v0",
+            license="MIT",
+            description="Unrolling example dataset",
+        )
+
+        self._benchmarks = {
+            "benchmark://unrolling-v0/offsets1": Benchmark.from_file_contents(  # FIXME: "/offsets1": Benchmark.from_file_contents(
```

Comment:
I think this code should work fine against stable. The problem was the example version in stable has a bug. Line 77 should be `return self._benchmarks[uri.path]`:

https://github.com/facebookresearch/CompilerGym/blob/stable/examples/example_compiler_gym_service/__init__.py#L64-L77